packages/node/src/transports/base.ts

```diff
@@ -123,13 +141,74 @@ export abstract class BaseTransport implements Transport {
+  protected _disabledUntil(requestType: SentryRequestType): Date {
+    const category = CATEGORY_MAPPING[requestType];
+    return this._rateLimits[category] || this._rateLimits.all;
+  }
+
+  /**
+   * Checks if a category is rate limited
+   */
+  protected _isRateLimited(requestType: SentryRequestType): boolean {
+    return this._disabledUntil(requestType) > new Date(Date.now());
+  }
+
+  /**
+   * Sets internal _rateLimits from incoming headers. Returns true if headers contains a non-empty rate limiting header.
+   */
+  protected _handleRateLimit(headers: Record<string, string | null>): boolean {
+    const now = Date.now();
+    const rlHeader = headers['x-sentry-rate-limits'];
+    const raHeader = headers['retry-after'];
+
+    if (rlHeader) {
+      // rate limit headers are of the form
+      //     <header>,<header>,..
+      // where each <header> is of the form
+      //     <retry_after>: <categories>: <scope>: <reason_code>
+      // where
+      //     <retry_after> is a delay in ms
+      //     <categories> is the event type(s) (error, transaction, etc) being rate limited and is of the form
+      //         <category>;<category>;...
+      //     <scope> is what's being limited (org, project, or key) - ignored by SDK
+      //     <reason_code> is an arbitrary string like "org_quota" - ignored by SDK
+      for (const limit of rlHeader.trim().split(',')) {
+        const parameters = limit.split(':', 2);
+        const headerDelay = parseInt(parameters[0], 10);
+        const delay = (!isNaN(headerDelay) ? headerDelay : 60) * 1000; // 60sec default
+        for (const category of (parameters[1] && parameters[1].split(';')) || ['all']) {
+          // categoriesAllowed is added here to ensure we are only storing rate limits for categories we support in this
+          // sdk and any categories that are not supported will not be added redundantly to the rateLimits object
+          const categoriesAllowed = [
+            ...(Object.keys(CATEGORY_MAPPING) as [SentryRequestType]).map(k => CATEGORY_MAPPING[k]),
+            'all',
+          ];
+          if (categoriesAllowed.includes(category)) this._rateLimits[category] = new Date(now + delay);
+        }
+      }
+      return true;
+    } else if (raHeader) {
+      this._rateLimits.all = new Date(now + parseRetryAfterHeader(now, raHeader));
+      return true;
+    }
+    return false;
+  }
+
   /** JSDoc */
-  protected async _send(sentryReq: SentryRequest): Promise<Response> {
+  protected async _send(sentryReq: SentryRequest, originalPayload?: Event | Session): Promise<Response> {
     if (!this.module) {
       throw new SentryError('No module available');
     }
-    if (new Date(Date.now()) < this._disabledUntil) {
-      return Promise.reject(new SentryError(`Transport locked till ${this._disabledUntil} due to too many requests.`));
+    if (originalPayload && this._isRateLimited(sentryReq.type)) {
+      return Promise.reject({
+        event: originalPayload,
```

Comment:
Why do we include the original payload here?

It is strange that the object key is `event` but the value is either an Event or a Session.